manual/sphinx/user_docs/notes.rst

```diff
@@ -1,18 +1,32 @@
 Notes
 =====
 
+Configure options
+-----------------
+
+Configure with ``--enable-checks=3`` enables a lot of checks of
+operations performed by the field objects. This is very useful for
+debugging a code, and can be omitted once bugs have been removed.
+``--enable=checks=2`` enables less checking, especially the
+computationally rather expensive ones, while ``--enable-checks=0``
+disables most checks.
+
+To get most checking, both from BOUT and from the compiler
+``--enable-debug`` can be used. That enables checks of level 3, as
+well as debug flugs, e.g. ``-g`` for gcc.
+
+For (sometimes) more useful error messages, there is the
+``--enable-track`` option. This keeps track of the names of variables
+and includes these in error messages.
+
+To enable opimization, configure with ``--enable-optimize=3``.
```

Comment:
The `--enable-optimize` flag is actually speculative: it tries to compile a test program with a few different flags, keeping any that the compiler recognises. It might not actually enable any optimisation depending on the compiler you're using.

Maybe we should note that there are also 3 levels of optimisation?